include/nil/crypto3/zk/snark/systems/plonk/placeholder/verifier.hpp

```diff
@@ -147,10 +147,17 @@ namespace nil {
                         auto numerator = challenge.pow(preprocessed_public_data.common_data.rows_amount) - FieldType::value_type::one();
                         numerator /= typename FieldType::value_type(preprocessed_public_data.common_data.rows_amount);
 
+                        // If public input sizes are set, all of them should be set.
+                        if(constraint_system.public_input_sizes_num() != 0 && constraint_system.public_input_sizes_num() != table_description.public_input_columns){
+                            return false;
+                        }
+
                         for( std::size_t i = 0; i < public_input.size(); ++i ){
                             typename FieldType::value_type value = FieldType::value_type::zero();
+                            std::size_t max_size = constraint_system.public_input_sizes_num() == 0 ? public_input[i].size() : constraint_system.public_input_size(i);
```

Comment:
I don't get this code, you have public_input[i].size() both here and below. So looks like it's better to write

std::size_t max_size = public_input[i].size();
if (constraint_system.public_input_sizes_num() != 0)
   max_size = std::min(max_size, constraint_system.public_input_size(i));